demo-server/src/test/java/com/illia/FileHolderTest.java

```diff
@@ -0,0 +1,34 @@
+package com.illia;
+
+import com.illia.server.file_holder.FileHolder;
+import com.illia.server.file_holder.FileHolderImpl;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.*;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest(classes = {FileHolderImpl.class})
+public class FileHolderTest {
+
+    @Autowired
+    FileHolder fileHolder;
+
+    @Test
+    public void testFileHolder() throws URISyntaxException {
```

Comment:
this is almost end-to-end test, which used too, but is much heavier for executing.
Try writing unit tests (maybe for validation service that I wrote above)
- there you want to test single class method, validator.validate(content)
also, write some controller tests. There we will be working with mocks (read about mockito, mocking and stubbing)
- try calling a controller method as you did here, but instead of checking a result, check if methods you wanted to be called were actually invoked. 

That way, this test can be split into two, smaller ones that will show immediately where the problem is, instead of failing a big one, and looking for the exact place in the debugger